### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by aleksandricbm**
Codebreaker-Rack-App
https://codebreaker-aleksandricbm.herokuapp.com/

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source "https://rubygems.org"
+gem 'rack'
+gem 'codebreaker', :git => 'https://github.com/aleksandricbm/Codebreaker.git', :branch => "processing"
```
Comment (kirillshevch) [inline Gemfile]: please, use new syntax:
```
gem 'codebreaker', git: 'https://github.com/aleksandricbm/Codebreaker.git', branch: 'processing'
```

File: config.ru [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './lib/racker'
+
+Racks = Rack::Builder.new do
+  use Rack::Static, urls: ['/stylesheets'], root: 'public'
+  use Rack::Session::Cookie, :key => 'rack.session',
+                             :path => '/',
+                             :expire_after => 2592000,
+                             :secret => 'change_me'
```
Comment (kirillshevch) [inline config.ru]: the same

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  def initialize(user_name)
+      @game = Codebreaker::Game.new
+      @game.start
+      @result_answer = []
+      @user_name = user_name
+      @attempts = ATTEMPT_COUNT
+  end
+
+  def save_object_game(object,file_name)
+    binding.pry
```
Comment (kirillshevch) [inline lib/game.rb]: ?

File: lib/game.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+class GameRacker
+  attr_accessor :attempts
+  ATTEMPT_COUNT = 5
+
+  def initialize(user_name)
+      @game = Codebreaker::Game.new
+      @game.start
+      @result_answer = []
+      @user_name = user_name
+      @attempts = ATTEMPT_COUNT
```
Comment (kirillshevch) [inline lib/game.rb]: use 2 spaces instead of 4

File: file_session/10a23a797ef234f076042f55dcaa3e49bf33ddd9346a74ceea8e2460a071468c [hunk 5]
```diff
@@ -0,0 +1,7 @@
+--- !ruby/object:GameRacker
+game: !ruby/object:Codebreaker::Game
+  secret_code: '6561'
+  result: ''
+result_answer: []
+user_name: jjj
+attempts: 5
```
Comment (kirillshevch) [inline file_session/10a23a797ef234f076042f55dcaa3e49bf33ddd9346a74ceea8e2460a071468c]: add /file_session/* to gitignore

File: score.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+--- 'Test : [["1234", "+"], ["1234", "+"], ["1234", "+"], ["1234", "+"]]'
+--- '2222 : [["1234", "++"], ["1234", "++"], ["1234", "++"], ["1234", "++"]]'
+--- ' : [["1234", "+--"], ["1234", "+--"], ["1234", "+--"], ["1234", "+--"]]'
+--- 'sasha : [["2462", "++++"]]'
+--- ' : [["3565", "++++"]]'
+--- ' : [["1234", "--"], ["1234", "--"], ["1234", "--"], ["1234", "--"]]'
+--- 'Sasha : [["1234", "---"], ["1234", "---"], ["1234", "---"], ["1234", "---"]]'
+--- ' : [["1234", "---"], ["1234", "---"], ["1234", "---"], ["1234", "---"]]'
+--- ' : [["1234", "---"], ["1234", "---"], ["1234", "---"], ["1234", "---"]]'
+--- 'рпар : [["1234", "+-"], ["4422", "-"], ["1234", "+-"], ["4156", "-"]]'
```
Comment (kirillshevch) [inline score.yaml]: add it to .gitignore

**Comment by kirillshevch** [comment]
hint feature?

File: config.ru [hunk 7]
```diff
@@ -0,0 +1,12 @@
+require './lib/racker'
+
+Racks = Rack::Builder.new do
```
Comment (kirillshevch) [inline config.ru]: app = Rack::Builder.new do

File: config.ru [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  use Rack::Static, urls: ['/stylesheets'], root: 'public'
+  use Rack::Session::Cookie, :key => 'rack.session',
+                             :path => '/',
+                             :expire_after => 2592000,
+                             :secret => 'change_me'
+  run Racker
+end
+
+run Racks
```
Comment (kirillshevch) [inline config.ru]: run app

File: lib/racker.rb [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+  def initialize(env)
+    @request = Rack::Request.new(env)
+    @request.session["init"] = true
+    @session_id = @request.session['session_id']
+    @file_name = File.expand_path("../../#{PATH}#{@session_id}", __FILE__)
+    init_session
+  end
+
+  def init_session
```
Comment (kirillshevch) [inline lib/racker.rb]: private

File: views/foother.html.erb [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
Comment (kirillshevch) [inline views/foother.html.erb]: ?

Comment (aleksandricbm) [inline views/foother.html.erb]: unnecessary template, fix

**Pull request merged**